### DISCUSSION TITLE
help for sorting by file type then by date added

### DISCUSSION THREAD

**Discussion by imed-ghomari** [SebastianMC/obsidian-custom-sort#71] [Q&A] [answered]
I have this file structure 
```
folder 1
   folder 1.1
       markdown
       pdf
       word
       markdown
   folder 1.2
folder 2
...
```
I want to sort the folder 1.1 and 1.2 by date of creation

And I want to sort the files within these folder by file type, so markdown files up top, then at the bottom all other file types. And I want this sorting to apply to all folders within the folder 1.1 and 1.2 if there are some.

Is this possible ?

**Comment by SebastianMC** [comment, marked as answer]
Hi @imeed166 

at a glance everything you described can be configured.
Let me go step by step:

### Push markdown files to the top

By default the matching is done against folder and file names, ignoring the file extension. \
If you want to match against the full file name (name + extension) you use the `/:files.` syntax (or the abbreviated variant `/:.`)
For example:

```
  /:files. ....md
```

The little strange `....md` above is a result of concatenation of `...` and `.md` and it means _'any file name ending with .md'_

If a sorting specification for a folder comprises only the above line it will push to the top the markdown files, the rest of folder items (sub-folders and other files) will go below. No additional order specification will automatically apply the alphabetical order both to the markdown files group (pushed to the top) and to the remaining folder items

### Applying a sorting specification to multiple folders and their subfolders

A sorting specification can be applied to multiple target folders. Also, a specific syntax exists to express _'the folder and its direct sub-folders'_ (`/...`) or _'the folder and all of its descendant folders'_ (`/*`).
For example:

```
  target-folder: folder 1/folder 1.1/*
  target-folder: folder 1/folder 1.2/*
```

specifies as a target the entire sub-tree of folders originating from `folder 1/folder 1.1` and `folder 1/folder 1.2`

### Sorting by date of creation

You wrote:
> I want to sort the folder 1.1 and 1.2 by date of creation

it is unclear for me if it means:
- (A) _sort all items inside the 'folder 1.1' and 'folder 1.2' by date of creation_
or
- (B) _if the 'folder 1.1' was created later than 'folder 1.2' then let it be displayed above 'folder 1.2' inside 'folder 1'_

If (A) is the case then the full sorting specification would look like:

```yaml
sorting-spec: |
  target-folder: folder 1/folder 1.1/*
  target-folder: folder 1/folder 1.2/*
  > advanced created
  /:files. ....md
```

which can be read as: _for all folders in the subtree of folders 'folder 1/folder 1.1/' and 'folder 1/folder 1.2/' the default sorting should be by 'creation date, advanced, descending (newest item in the top)'. Then in each folder in these subtrees the markdown files to be pushed to the top._

If (B) is the case then the full sorting specification would look like:

```yaml
sorting-spec: |
  target-folder: folder 1  
  > advanced created
  target-folder: folder 1/folder 1.1/*
  target-folder: folder 1/folder 1.2/*
  /:files. ....md
```

which can be read as: _sort items in the folder 'folder 1' by 'creation date, advanced, descending (newest item in the top)'. And in all folders in the subtree of folders 'folder 1/folder 1.1/' and 'folder 1/folder 1.2/' push markdown files to the top._

> REMARK: because folders in Obsidian don't carry date attributes (e.g. creation date or modification date) the possible workaround is to take the date of files in the folder (e.g. date of the oldest file for advanced sorting by creation date). That's what the `advanced created` does.

### Final remarks

I hope I correctly guessed your intentions behind the question.
If not, feel free to continue here

**Reply by imed-ghomari** [reply]
Thank you for the detailed explanation, I appreciate it.

I was talking about the (B) use case.

and I only have one final question. The folders 1.1 and 1.2 are always changing so I don't want to specify them in the sorting, in this case can I change the specification to this ? 
```YAML
sorting-spec: |
  target-folder: folder 1  
  > advanced created
  target-folder: folder 1/*
  /:files. ....md
```

**Reply by imed-ghomari** [reply]
I tried it, and it works, thank you again.

**Comment by imed-ghomari** [comment]
I'm actually using `< advanced created` to sort the folders that were created earlier at the top.

> (e.g. date of the oldest file for advanced sorting by creation date). That's what the advanced created does.

I have a folder that already has a markdown file in it. When I create another markdown file in it, the folder moves to the bottom. Is this normal behaviour? Shouldn't the folder stay in the same place because the file I created is new and the sort should follow the oldest file in the folder like you said?

If I need to create an issue or another discussion for this, tell me, I'll move it there.

**Reply by SebastianMC** [reply]
At a glance the described behavior is unexpected. Let me try to reproduce it on my side, then I will let know

**Reply by SebastianMC** [reply]
This is a bug indeed :-)
I created a new ticket #72 with reference to your comment.
The fix will be available in a new release in a few minutes

Thanks for catching the problem and reporting!!!

BTW if you prefer to report the bug in a new ticket by yourself, feel free to do it. I'm aware that collecting stats on github matters to some persons.

**Reply by SebastianMC** [reply]
The release with the fix is [1.7.2](https://github.com/SebastianMC/obsidian-custom-sort/releases/tag/1.7.2)

**Reply by imed-ghomari** [reply]
no I don't care about github stats, I confirm that it works now, thank you for the quick fix.
